### ISSUE TITLE
Copy/paste broken in Vim/Tmux + Text selection broken in tmux

### ISSUE THREAD

**Issue by flybayer** [zeit/hyper#773]
Clicking and dragging on text in tmux, such as a directory listing, does nothing.

Copy & paste also does nothing in tmux. I can paste all day long in a non-tmux pane but never in a tmux pane.

This is a show-stopper for me :sob: 

_I'm on Ubuntu 14.04_

**Comment by jonsterling** [comment]
I'll add that this *used* to work, but it is no longer working.

**Comment by seagoj** [comment]
I agree. I've used hyper/hyperterm daily for a couple of months now and it was basically a total replacement for iterm2 for me, but this is a showstopper. Tmux is an essential part of my workflow.

**Comment by seagoj** [comment]
With the console open, whenever I move the mouse in tmux, I get hundreds of errors similar to the following:

**bundle.js:34037 Uncaught InvalidStateError: Failed to execute 'collapseToEnd' on 'Selection': there is no selection.hterm.Terminal.onMouse_ @ bundle.js:34037**

**Comment by bradparks** [comment]
Ditto here - this used to work, but no longer works. Can't copy paste, and I can't navigate with the mouse to Tmux panes either. 

- I tried bringing the session active in iTerm2, and it works fine (paste and pane navigation). 
- go back to hyper.app, fails (nothing pasted)
- try in hyper.app without tmux, paste works.

**Comment by flybayer** [comment]
The breaking commit is a7595c1a45bf584c9baee490e6c3bffc04732183 (split-panes)

**Comment by flybayer** [comment]
@ekmartin why did you add this code to `lib/hterm.js`? This section is the issue cause. Everything seems to work fine when I take it out.

```js
// sine above we're no longer relying on `preventDefault`
// to avoid selections, we use css instead, so that
// focus is not lost, but selections are still not possible
// when the appropiate VT mode is set
hterm.VT.prototype.__defineSetter__('mouseReport', function (val) {
  this.mouseReport_ = val;
  const rowNodes = this.terminal.scrollPort_.rowNodes_;
  if (rowNodes) {
    if (val === this.MOUSE_REPORT_DISABLED) {
      rowNodes.style.webkitUserSelect = 'text';
    } else {
      rowNodes.style.webkitUserSelect = 'none';
    }
  }
});

hterm.VT.prototype.__defineGetter__('mouseReport', function () {
  return this.mouseReport_;
});
```
from here: https://github.com/zeit/hyper/blob/master/lib/hterm.js#L210-L228

**Comment by flybayer** [comment]
:point_up_2: that's also the cause of #789

**Comment by ekmartin** [comment]
@flybayer: @rauchg [implemented](https://github.com/zeit/hyper/pull/693/commits/a4315042fa3266d42b67e74f9ac3403ab7e3ca16) that to fix an issue where applications such as htop weren't getting focused correctly when clicked. Seems like we'll have to look for another work around though!

**Comment by flybayer** [comment]
@ekmartin would it be ok to revert this for now? I think more people want to paste with tmux/vim than use `htop` :smile:

**Comment by ekmartin** [comment]
Yes, I agree that makes sense. Do you want to send a PR?

**Comment by ekmartin** [comment]
This has been fixed in 0.8.2.
